File: sdk/python/hitechcloud/resources/location_v2.py
"""LocationV2 resource (4 endpoints)"""

from .base import BaseResource

class LocationV2Resource(BaseResource):
    """LocationV2: countries, states, cities lookup"""

    def list_countries(self) -> dict:
        """GET /api/location_v2/countries - Return list of countries"""
        return self._http.get("/api/location_v2/countries")

    def get_states(self, country_code: str) -> dict:
        """GET /api/location_v2/{code}/state - Return states for a country code"""
        return self._http.get(f"/api/location_v2/{country_code}/state")

    def get_cities(self, state_id: int) -> dict:
        """GET /api/location_v2/state/{id}/city - Return cities for a state ID"""
        return self._http.get(f"/api/location_v2/state/{state_id}/city")

    def get_cities_by_state_name(self, name: str) -> dict:
        """GET /api/location_v2/state/city/{name} - Return cities for a state name"""
        return self._http.get(f"/api/location_v2/state/city/{name}")

File: sdk/ruby/lib/hitechcloud/resources/location_v2.rb
module HiTechCloud
  # LocationV2 resource
  class LocationV2Resource < BaseResource

    # Get countries
    def countries
      @http.get("/api/location_v2/countries")
    end

    # Get states
    def states(code)
      @http.get("/api/location_v2/%{code}/state" % { code: code })
    end

    # Get cities
    def cities(id)
      @http.get("/api/location_v2/state/%{id}/city" % { id: id })
    end

    # Get city by name
    def city_by_name(name)
      @http.get("/api/location_v2/state/city/%{name}" % { name: name })
    end

  end
end
